Pocket/pocket-ios
# Core Data - Use child contexts to fetch archived items and tags

## Summary
* This PR adds to the work done in #570 to apply the same fetch approach to archived items and tags

## Implementation Details
* Add `DerivedSpace`, a type that handles save operations on paginated data (saved items, archived items and tags)
* Update `FetchArchive`, `FetchSaves` and `FetchTags` to use `DerivedSpace`
* Remove `SavedItemSpace`, replaced by `DerivedSpace`
* Make sure the right context is injected in the all the calls originating from `DerivedSpace`

## Test Steps
* Use the app, particularly log out/log in with accounts that have a large number of items, and make sure that:
    - there are no crashes
    - the ui stays responsive

## PR Checklist:
